plugins/vli/fu-vli-usbhub-device.c

```diff
@@ -412,6 +424,42 @@ fu_vli_usbhub_device_attach(FuDevice *device, FuProgress *progress, GError **err
 						    tmp ^ (1 << 1),
 						    error))
 			return FALSE;
+	} else if (fu_device_has_private_flag(device,
+					      FU_VLI_USBHUB_DEVICE_FLAG_ATTACH_WITH_USB_CABLE)) {
+		g_autoptr(FwupdRequest) request = fwupd_request_new();
+
+		/* the user has to do something */
+		fwupd_request_set_kind(request, FWUPD_REQUEST_KIND_IMMEDIATE);
+		fwupd_request_set_id(request, FWUPD_REQUEST_ID_REMOVE_REPLUG);
+		fwupd_request_add_flag(request, FWUPD_REQUEST_FLAG_ALLOW_GENERIC_MESSAGE);
+		fwupd_request_set_message(
+		    request,
+		    "The update will continue when the device USB cable has been "
+		    "unplugged and then re-inserted.");
+		if (!fu_device_emit_request(device, request, progress, error))
+			return FALSE;
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+
+		/* success */
+		return TRUE;
+	} else if (fu_device_has_private_flag(device,
+					      FU_VLI_USBHUB_DEVICE_FLAG_ATTACH_WITH_POWER_CORD)) {
+		g_autoptr(FwupdRequest) request = fwupd_request_new();
+
+		/* the user has to do something */
+		fwupd_request_set_kind(request, FWUPD_REQUEST_KIND_IMMEDIATE);
+		fwupd_request_set_id(request, FWUPD_REQUEST_ID_REMOVE_REPLUG);
+		fwupd_request_add_flag(request, FWUPD_REQUEST_FLAG_ALLOW_GENERIC_MESSAGE);
```

Comment:
Te way generic message works is that we don't show the text set with fwupd_request_set_message() and instead show something translated based off the enum sent in fwupd_request_set_id(). In this case I do think we need to use generic, but I think we need to **add** another request ID, e.g. FWUPD_REQUEST_ID_REPLUG_POWER with the description of something like "The update will continue when the device power cable has been unplugged and then re-inserted" -- can you add that to this PR please.

Reply:
Actually, I did that for you in https://github.com/fwupd/fwupd/pull/6403

Reply:
Thank you very much for your help.